configuration/salt.md

```diff
@@ -53,6 +55,123 @@ function that, when issued, will immediately halt the computer. There is another
+to order the execution of states.
+`cmd.run` is the execution module. It decides which action will be executed.
+`name: echo 'hello world'` is a parameter for the execution module. It depends on
+the module which parameters are accepted.
+
+There is list of [officially available states][salt-doc-states].
+There are many very useful states:
+* For [managing files][salt-doc-states-file]: Use this to create files or 
+  directories and change them (append lines, replace text, set their content etc.)
+* For [installing and uninstalling][salt-doc-states-pkg] packages.
+* To [execute shell commands][salt-doc-states-cmd].
+
+With these three states you can do most of the configuration inside of a vm.
+
+You also can [order the execution][salt-doc-states-order] of your states:
+
+    D:
+      cmd.run:
+      - name: echo 1
+      - order: last
+    C:
+      cmd.run:
+      - name: echo 1
+    B:
+      cmd.run:
+      - name: echo 1
+      - require:
+        - cmd: A
+      - require_in:
+        - cmd:C
+    A:
+      cmd.run:
+      - name: echo 1
+      - order: 1
+
+The order of execution will be `A, B, C, D`.
+The official documentation has more details on the [require][salt-doc-states-req] and
+[order][salt-doc-states-ord] arguments.
+
+#### State files
+
+When configuring a system you will write one or several state files (`*.sls`) and
+put (or symlink) them in the salt main directory `/srv/salt/`.
+Each state file contains one multiple states and should describe some unit of 
+configuration (e.g.: A state file `mail.sls` could setup a vm for mailing).
+
+#### Top files
+
+After you have state several state files, you need something to assign them to a
+vm. This is done by `*.top` files ([official documentation][salt-doc-top]).
+Their structure looks like this:
+
+    environment:
+      target_matching_clause:
+      - statefile1
+      - folder2.statefile2
+
+The environment will be in most cases `base`.
+The `target_matching_clause` will be used to select your minions (vms).
+It can be just the name of a vm or a regular expression.
+If you are using a regular expression, you need to give salt a hint you are doing
+so:
+
+    environment:
+      ^app-(work|(?!mail).*)$:
+      - match: pcre
+      - statefile
+
+For each target you can write a list of state files.
+Each line is a path to a state file (without the `.sls`) relative to the main 
+directory. Each `/` is exchanged by a dot, so you can't reference files or 
+directories with a dot in their name.
+
+### Enabling top files and applying the configuration
+
+Now because we use custom extension to manage top files (instead of just
+enabling them all) to enable the particular top file you should issue command:
+
+    qubesctl top.enable my-new-vm
+
+To list all enabled tops:
+
+    qubesctl top.enabled
+
+And to disable one:
+
+    qubesctl top.disable my-new-vm
+
+To actually apply the state:
+
+    qubesctl state.highstate
```

Comment:
IMO it's better to point `qubesctl --all state.highstate` here, as you talk about configuring VMs and without `--all` option only dom0 will be configured.